### PR TITLE
Fix dApp AddEthereumChain requests

### PR DESCRIPTION
**Pull request by yrliou**
- Check URL params are HTTPS or HTTP localhost URLs
- Only issue eth_chainId RPC call after user approval

Resolves https://github.com/brave/brave-browser/issues/24253

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
See https://github.com/brave/internal/issues/885#issuecomment-1194508218

### CODE REVIEW

**Comment by thypon** [comment]
Is there any build link ready here?

**Comment by yrliou** [comment]
> Is there any build link ready here?

@thypon Not yet, CI is currently building, I'll ping you with a link when the PR build can be downloaded.

File: components/brave_wallet/browser/ethereum_provider_impl.cc [hunk 1]
```diff
@@ -158,7 +158,7 @@ void EthereumProviderImpl::AddEthereumChain(const std::string& json_payload,
                             "", true);
     return;
   }
-  auto chain = brave_wallet::ValueToEthNetworkInfo(*list.begin());
+  auto chain = brave_wallet::ValueToEthNetworkInfo(*list.begin(), true);
```
Comment (supermassive) [inline components/brave_wallet/browser/ethereum_provider_impl.cc:161]: This new filtering perfectly fits separate method for Value -> NetworkInfo conversion I've made for parsing wallet_addEthereumChain payload in a pending PR [here ](https://github.com/brave/brave-core/pull/14104/files#diff-46ae066e01b73c7dfe02c4a519a7a77b8f4a66c4e230ea2301e4fd84cca322a1R98) 
The problem was that brave's and Eip3085 base::Value view of NetworkInfo started to differ very much. 
Now I will have to fix conflicts precisely 🙃

**Review by bbondy — APPROVED** [review]

**Comment by yrliou** [comment]
https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-fix_add_ethereum_chain-windows/detail/pr-brave-browser-fix_add_ethereum_chain-windows/7/pipeline/311
Windows build is green from start to test-browser, but it is still running test-browser-chromium, and has dist step pending. Since this PR is not platform dependent with all other platforms passed including window-x86, and this PR won't have any effects on those later stages anyway, I'm merging it now instead of waiting for another 1.5hr.

**Comment by srirambv** [comment]
Verification passed on  
<html>
<body>


Brave | 1.44.3 Chromium: 104.0.5112.57 (Official Build) nightly (64-bit)
-- | --
Revision | `212fd173a0da1e0a024f328295bb56a2529190bb-refs/branch-heads/5112@{#1042}`
OS | Windows 11 Version 21H2 (Build 22000.795)


</body>
</html>
- Verification notes can be found [here](https://github.com/brave/internal/issues/885#issuecomment-1197656999)

**Pull request merged**